vent/coordinator/ipc.py

```diff
@@ -1,28 +1,90 @@
 import threading
 import time
+from enum import Enum, auto
+import pickle
+import socket
+
+buffer_size = 65536
+
+
+class IPCCommand(Enum):
+    START = auto()
+    STOP = auto()
+    GET_SENSORS = auto()
+    GET_CONTROL = auto()
+    SET_CONTROL = auto()
+    GET_ACTIVE_ALARMS = auto()
+    GET_LOGGED_ALARMS = auto()
+    CLEAR_LOGGED_ALARMS = auto()
+
+
+class IPCMessage:
+    def __init__(self, command, args=None):
+        """
+        :param command: ENUM in IPCCommand
+        """
+        self.command = command
+        # TODO: current only assume one arg
+        self.args = args
 
 
 class IPC:
     # Class for communicating between processes either by sockets
     # or named-pipes
     #   Functions:
-    def __init__(self, listen=False):
+    def __init__(self, listen, addr=None, port=None):
+        self.listen = listen
+        self.s = socket.socket()
+        if listen:
+            # TODO: can start random port on testing, but deterministic port on deployment
+            self.s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            self.s.bind(('', 0))
```

Comment:
if this is binding to port 0 imo that makes it harder than it needs to be to identify, but i suppose we can fix this if it becomes a problem rather than preoptimizing. Not sure of socket syntax and whether `''` binds to `'localhost'`.

Reply:
Changing to RPC solves this issue.